Add doesNotHaveToString assertion

lykims opened the pull request:
#### Check List:
* Fixes #2028 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

Minor fixes for `hasToString` doc description and `ShouldHaveToString` error as well.

joel-costigliola commented on src/main/java/org/assertj/core/api/Assert.java:
```diff
@@ -467,11 +467,30 @@
+   * // Instead of writing ...
+   * assertThat(homer.toString()).isNotEqualTo("Marge");
+   * // ... you can simply write:
+   * assertThat(homer).doesNotHaveToString("Marge");
+   * </code></pre>
```
join the two lines

joel-costigliola commented on src/main/java/org/assertj/core/error/ShouldNotHaveToString.java:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  private ShouldNotHaveToString(Object actual, String other) {
+    super("%nExpecting actual's toString() not to be equal to:%n  <%s>%nbut was:%n  <%s>", other, actual.toString());
```
let's simplify this to:
```
"%nExpecting actual's toString() not to be equal to:%n  <%s>"
```
the reason is `but was:%n  <%s>` section is actually repeating actual's `toString` value

joel-costigliola commented on src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:
```diff
@@ -0,0 +1,65 @@
+    // GIVEN
+    Object actualObject = null;
+    // WHEN
+    ThrowingCallable code = () -> objects.assertDoesNotHaveToString(someInfo(), actualObject, "bar");
```
use `expectAssertionError` from `org.assertj.core.util.AssertionsUtil` which returns an `AssertionError`

joel-costigliola commented on src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:
```diff
@@ -0,0 +1,65 @@
+    AssertionError error = expectAssertionError(() -> objects.assertDoesNotHaveToString(info, actual, "Person[name='foo']"));
+    // THEN
+    verify(failures).failure(info, shouldNotHaveToString("Person[name='foo']", "Person[name='foo']"));
+    assertThat(error).hasMessageContainingAll("Person[name='foo']");
```
remove this line, it is already covered by `verify(failures)` (we don't want to test the error message but only that we call the proper error message factory)

joel-costigliola commented on src/test/java/org/assertj/core/internal/objects/Objects_assertDoesNotHaveToString_Test.java:
```diff
@@ -0,0 +1,65 @@
+    // GIVEN
+    AssertionInfo info = someInfo();
+    // WHEN
+    AssertionError error = expectAssertionError(() -> objects.assertDoesNotHaveToString(info, actual, "Person[name='foo']"));
```
remove `AssertionError error` local variable (see next comment why)

joel-costigliola reviewed (changes requested):
@lykims thanks, good PR, just a few improvements to do and we are good.

lykims commented:
@joel-costigliola I addressed your comments. I also applied changes suggested for `Objects_assertDoesNotHaveToString_Test` to `Objects_assertHasToString_Test`. Let me know if I miss anything!

joel-costigliola commented:
Integrated thanks for the good work @lykims!
